src/e3/pytest.py

```diff
@@ -0,0 +1,235 @@
+        return
+
+    global test_errors
+    if test_errors:
+        # Return with an exit code of `3` if we encountered errors (not failures).
+        # This is the exit code that corresponds to an "internal error" according to the
+        # pytest docs, which is the closest thing to having an actual Python error in
+        # test code.
+        session.exitstatus = 3
+
+    if session.config.getoption("cov_source"):
+        cov_file = str(session.config.rootpath / ".coverage")
+        if session.config.getoption("e3_cov_rewrite"):
+            origin_dir, new_dir = session.config.getoption("e3_cov_rewrite")
+            fix_coverage_paths(origin_dir=origin_dir, new_dir=new_dir, cov_db=cov_file)
+
+        os_name = Env().build.os.name
+
+        cov = Coverage(data_file=cov_file)
+
+        # Exclude all lines matching the DEFAULT_EXCLUDE_LIST
+        for regex in DEFAULT_EXCLUDE_LIST:
+            cov.exclude(regex)
+
+        # Exclude all <os>-only line (unless that's for this OS)
+        for regex in (
+            "%s-only" % o
+            for o in ("darwin", "linux", "solaris", "windows", "bsd", "aix")
+            if o != os_name
+        ):
+            cov.exclude(regex)
+
+        # Handling of Windows
+        if os_name == "windows":
+            cov.exclude('if sys.platform != "win32":')
+        else:
+            cov.exclude('if sys.platform == "win32":')
+            cov.exclude("unix: no cover")
+
+        # Exclude no cover for this OS
+        cov.exclude(f"{os_name}: no cover")
+
+        cov.load()
+
+        # Read configuration files in <root>/tests/coverage to fix the list
+        # of files to omit for OS specific list
+        # We're relying on the default coverage configuration for the
+        # platform-agnostic list
+        omit_files: list[str] = cov.get_option("run:omit") or []  # type: ignore
+        coverage_conf_dir = session.config.rootpath / "tests" / "coverage"
+        conf_file = coverage_conf_dir / f"omit-files-{os_name}"
+        if conf_file.exists():
+            with conf_file.open() as f:
+                for line in f:
+                    omit_files.append(line.rstrip())
+
+        # cov.html_report(directory=str(session.config.rootpath ), omit=omit_files)
+        cov.html_report(omit=omit_files)
+        cov.report(omit=omit_files, precision=3)
+        cov.xml_report()
+
+
+def fix_coverage_paths(origin_dir: str, new_dir: str, cov_db: str) -> None:
+    """Fix coverage paths.
+
+    :param origin_dir: path to the package directory, e.g.
+        .tox/py311-cov-xdist/lib/python3.11/site-packages/e3
+    :param new_dir: path to the dir that should be visible instead of origin_dir
+        e.g. src/
+    :param cov_db: path to the .coverage data base
+    """
+    paths = PathAliases()
+    paths.add(origin_dir, new_dir)
+
+    old_cov_file = NamedTemporaryFile(dir=os.path.dirname(cov_db))
+    old_cov_file.close()
+    try:
+        mv(cov_db, old_cov_file.name)
```

Comment:
Is it in the right way ?
We move `cov_db` to `old_cov_file.name` and finally remove `old_cov_file.name` ?

Reply:
What is the right way? ;) 

This is just moving the coverage file in another location and then deleting the extra file. Note that this is not a new code.